clients/client-python/gravitino/filesystem/gvfs.py

```diff
@@ -0,0 +1,667 @@
+            "/"
+        ):
+            raise GravitinoRuntimeException(
+                "Doesn't support copy a remote gvfs file to an another remote file."
+            )
+        context: FilesetContext = self._get_fileset_context(rpath)
+        context.get_fs().get_file(
+            self._strip_storage_protocol(
+                context.get_storage_type(), context.get_actual_path()
+            ),
+            lpath,
+            **kwargs,
+        )
+
+    def _convert_actual_path(self, path, context: FilesetContext):
+        """Convert an actual path to a virtual path.
+          The virtual path is like `fileset/{catalog}/{schema}/{fileset}/xxx`.
+        :param path: Actual path
+        :param context: Fileset context
+        :return A virtual path
+        """
+        if context.get_storage_type() == StorageType.HDFS:
+            actual_prefix = infer_storage_options(
+                context.get_fileset().storage_location()
+            )["path"]
+        elif context.get_storage_type() == StorageType.LOCAL:
+            actual_prefix = context.get_fileset().storage_location()[
+                len(f"{StorageType.LOCAL.value}:") :
+            ]
+        else:
+            raise GravitinoRuntimeException(
+                f"Storage type:{context.get_storage_type()} doesn't support now."
+            )
+
+        if not path.startswith(actual_prefix):
+            raise GravitinoRuntimeException(
+                f"Path {path} does not start with valid prefix {actual_prefix}."
+            )
+        virtual_location = self._get_virtual_location(context.get_name_identifier())
+        return f"{path.replace(actual_prefix, virtual_location)}"
+
+    def _convert_actual_info(self, entry: Dict, context: FilesetContext):
+        """Convert a file info from an actual entry to a virtual entry.
+        :param entry: A dict of the actual file info
+        :param context: Fileset context
+        :return A dict of the virtual file info
+        """
+        path = self._convert_actual_path(entry["name"], context)
+        return {
+            "name": path,
+            "size": entry["size"],
+            "type": entry["type"],
+            "mtime": entry["mtime"],
+        }
+
+    def _get_fileset_context(self, virtual_path: str):
+        """Get a fileset context from the cache or the Gravitino server
+        :param virtual_path: The virtual path
+        :return A fileset context
+        """
+        virtual_path: str = self._pre_process_path(virtual_path)
+        identifier: NameIdentifier = self._extract_identifier(virtual_path)
+        read_lock = self._cache_lock.gen_rlock()
+        try:
+            read_lock.acquire()
+            cache_value: Tuple[Fileset, AbstractFileSystem, StorageType] = (
+                self._cache.get(identifier)
+            )
+            if cache_value is not None:
+                actual_path = self._get_actual_path_by_ident(
```

Comment:
Is it possible that the `Fileset` of `FilesetContext` has been deleted and we can still get the deleted fileset info from the cache?

Reply:
Yes, I think it is possible. The purpose of introducing the cache is to reduce the access to Gravitino Server. In `Java GVFS`, we use `Caffine Cache`, a cache component that can automatically delete cache (Key, Value) after a certain period of access time, which can solve the cache expiration problem. I am not sure if there is a similar cache component in Python. Do you have anything about it?

Reply:
It seems that the current best practice is [cachetools](https://github.com/tkem/cachetools/), which provides `TTLCache` that adds the expiration time in cache.

Reply:
OK, thanks for sharing. Looks like it meets the needs, I’ll try to introduce this library.